plugins/tile/tile-plugin.cpp

```diff
@@ -271,25 +280,63 @@ class tile_output_plugin_t : public wf::pointer_interaction_t, public wf::custom
         });
     };
 
+    wf::activator_callback on_toggle_maximized = [=] (auto)
+    {
+        return conditioned_view_execute(true, [=] (wayfire_toplevel_view view)
+        {
+            auto node = tile::view_node_t::get_node(view);
+            if (!node)
+            {
+                return;
+            }
+
+            node->show_maximized = !node->show_maximized;
+            tile_workspace_set_data_t::get(view->get_wset())
+                .set_view_maximized(view, node->show_maximized);
+        });
+    };
+
     bool focus_adjacent(tile::split_insertion_t direction)
     {
         return conditioned_view_execute(true, [=] (wayfire_toplevel_view view)
         {
             auto adjacent = tile::find_first_view_in_direction(
                 tile::view_node_t::get_node(view), direction);
 
+            if (!adjacent)
+            {
+                return false;
+            }
+
+            auto current_node  = tile::view_node_t::get_node(view);
+            bool was_maximized = current_node->show_maximized;
+
+            if (was_maximized)
+            {
+                autocommit_transaction_t tx;
+                current_node->show_maximized = false;
+                current_node->set_geometry(current_node->geometry, tx.tx);
+            }
+
+            auto adjacent_node = tile::view_node_t::get_node(adjacent->view);
```

Comment:
This line is a bit confusing. `adjacent` is already a `view_node_t`. So you start with the node pointer (`adjacent`), get the view of it, then look for the node of this view. In other words, `adjacent` will always be the same as `adjacent_node`, no need for an extra lookup.

Reply:
When you realize this, you can actually merge the two `if`s together and use the same transaction for both `set_geometry` calls. The benefit is that by resizing both the old and the adjacent view with the same transaction, Wayfire will only start the animation when **both** views are ready for the resize.